### PR TITLE
[spaceship] add requests to remove or replace visible apps

### PR DESCRIPTION
**Pull request by lucgrabowski**
### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context

Add 2 requests to remove or replace visible apps for user.
Add a helper to `remove_users` to `App` model - similar to `add_users`

### Description

Changes for `User`:
- `delete_user_visible_apps` request added to remove apps from `User`
- `patch_user_visible_apps` request added to replace apps for `User`
- tests for both requests added

Changes in `App`: helper to `remove_users` (similar to `add_users`)

Just to mention, I didn't add a similar `replace_users` because it would require more requests inside to make it work properly:
1. fetch every user, 
2. for each of those users: 
a) check apps for this user, 
b). replace apps by removing only 1 app for user without removing others

And I'm not sure if it makes sense to add such a helper.

### Testing Steps

Tested in `irb`:
```ruby
require 'spaceship'
Spaceship::ConnectAPI.login
Spaceship::ConnectAPI.select_team

# testing `delete_user_visible_apps` and `patch_user_visible_apps`
apps=Spaceship::ConnectAPI::App.all
apps.collect { |a| a.bundle_id + " => " + a.id }

user=Spaceship::ConnectAPI::User.find(email: EMAIL").first
Spaceship::ConnectAPI.post_user_visible_apps(user_id: user.id,  app_ids: [APP_ID_1])
Spaceship::ConnectAPI.post_user_visible_apps(user_id: user.id,  app_ids: [APP_ID_2])

Spaceship::ConnectAPI.delete_user_visible_apps(user_id: user.id,  app_ids: [APP_ID_1])
Spaceship::ConnectAPI.patch_user_visible_apps(user_id: user.id,  app_ids: [APP_ID_1])

Spaceship::ConnectAPI.patch_user_visible_apps(user_id: user.id,  app_ids: [APP_ID_2])
Spaceship::ConnectAPI::User.find(email: EMAIL").first.visible_apps

# testing `remove_users`
app=Spaceship::ConnectAPI::App.find(BUNDLE_ID)
user=Spaceship::ConnectAPI::User.find(email: EMAIL").first
app.remove_users(user_ids: [user.id])
app.add_users(user_ids: [user.id])
```

### CODE REVIEW

File: spaceship/lib/spaceship/connect_api/users/users.rb [hunk 1]
```diff
@@ -42,6 +42,34 @@ def add_user_visible_apps(user_id: nil, app_ids: nil)
...
+                id: app_id
+              }
+            end
+          }
+
+          users_request_client.patch("users/#{user_id}/relationships/visibleApps", body)
+        end
+
+        # Remove app permissions for user
+        def remove_user_visible_apps(user_id: nil, app_ids: nil)
```
Comment (joshdholtz) [inline spaceship/lib/spaceship/connect_api/users/users.rb:60]:
```suggestion
        def delete_user_visible_apps(user_id: nil, app_ids: nil)
```

File: spaceship/lib/spaceship/connect_api/users/users.rb [hunk 2]
```diff
@@ -42,6 +42,34 @@ def add_user_visible_apps(user_id: nil, app_ids: nil)
           users_request_client.post("users/#{user_id}/relationships/visibleApps", body)
         end
 
+        # Replace app permissions for user
+        def replace_user_visible_apps(user_id: nil, app_ids: nil)
```
Comment (joshdholtz) [inline spaceship/lib/spaceship/connect_api/users/users.rb:46]:
```suggestion
        def patch_user_visible_apps(user_id: nil, app_ids: nil)
```

File: spaceship/lib/spaceship/connect_api/users/users.rb [hunk 3]
```diff
@@ -28,7 +28,7 @@ def delete_user(user_id: nil)
           users_request_client.delete("users/#{user_id}")
         end
 
-        # Change app permissions for user
+        # Add app permissions for user
         def add_user_visible_apps(user_id: nil, app_ids: nil)
```
Comment (joshdholtz) [inline spaceship/lib/spaceship/connect_api/users/users.rb:32]: We've been prefixing these methods with the HTTP method so I think I'd like to stick to that if possible 😇  I realized I should have caught this `add_` sooner so maybe update that to be deprecated and change it 🤷‍♂️  My apologies on that!

```suggestion
        # @deprecated Use {#post_user_visible_apps} instead.
        def add_user_visible_apps(user_id: nil, app_ids: nil)
          post_user_visible_apps(user_id: user_id, app_ids: app_ids)
        end

        def post_user_visible_apps(user_id: nil, app_ids: nil)
```

**Review by joshdholtz — CHANGES_REQUESTED** [review]
This looks good! Just a few naming tweaks 😊

**Comment by lucgrabowski** [comment]
@joshdholtz Thanks for review, I've applied changes and fixed tests.

**Comment by lucgrabowski** [comment]
@googlebot I fixed it.

**Comment by joshdholtz** [comment]
@googlebot I consent.

**Review by joshdholtz — APPROVED** [review]
This is working good for me! Thank you for adding this ❤️ Really appreciate the contribution!

**Pull request merged**